app/src/main/java/br/com/dionataferraz/vendas/data/local/TransactionDataSource.kt

```diff
@@ -0,0 +1,52 @@
+package br.com.dionataferraz.vendas.data.local
+
+import android.util.Log
+import br.com.dionataferraz.vendas.App
+import br.com.dionataferraz.vendas.data.local.entity.TransactionEntity
+import br.com.dionataferraz.vendas.viewmodel.Transaction
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class TransactionDataSource {
+
+    private val database: SalesDatabase by lazy {
+        SalesDatabase.getInstance(App.context)
```

Comment:
Com o ajuste do `SalesDatabase.kt`, podemos fazer dessa forma:
```suggestion
        SalesDatabase.getInstance()
```